src/changeColors.ts

```diff
@@ -0,0 +1,15 @@
+import * as vscode from 'vscode';
+import { getSetting } from './getColor';
+
+type ColorCustomization = { [key: string]: string | undefined }
+
+export default async (color?: string) => {
+    const settings = vscode.workspace.getConfiguration('workbench');
+    const currentColorCustomization: ColorCustomization = settings.get('colorCustomizations') || {};
+    const tabBarBorderColor = (getSetting('tabBorder') || true) ? color: undefined
```

Comment:
instead of `getSetting('tabBorder')` we can do `getSettings().tabBorder` and take advantage of TS. There is already a `Config` type, just need to expand it

Reply:
Now using getSettings().tabBorder: Moved getting settings to its own file getSettings.ts.

In order to get getSettings().config etc to work I had to rely on `unknown` type casting. I think this is needed because `vscode.workspace.getConfiguration('colorTabs')` returns vscode.WorkspaceConfiguration type which does not contain types needed. And `settings.get<T>(key)` returns only one settings at a time, as the key cannot be omitted. This way we are returning them all at once and thus getSettings().key can be used.